### PR TITLE
Core 7729 Allow users to download metadata template from metadata menu.

### PR DESCRIPTION
**Pull request by sri2k1us**

### CODE REVIEW

File: ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/SelectMetadataTemplateDialog.ui.xml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <!DOCTYPE ui:UiBinder SYSTEM "http://dl.google.com/gwt/DTD/xhtml.ent">
 <ui:UiBinder xmlns:ui='urn:ui:com.google.gwt.uibinder'
              xmlns:container="urn:import:com.sencha.gxt.widget.core.client.container"
-             xmlns:grid="urn:import:com.sencha.gxt.widget.core.client.grid">
+             xmlns:grid="urn:import:com.sencha.gxt.widget.core.client.grid"
+             xmlns:toolbar="urn:import:com.sencha.gxt.widget.core.client.toolbar"
+             xmlns:button="urn:import:com.sencha.gxt.widget.core.client.button">
```
Comment (ash3rz) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/SelectMetadataTemplateDialog.ui.xml]: Minor thing, but looks like changes were made here that weren't necessary?  It doesn't look like any toolbars or buttons are used in this file.

Comment (sri2k1us) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/metadata/dialogs/SelectMetadataTemplateDialog.ui.xml]: 👍

File: ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/file/DownloadController.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Created by sriram on 6/28/16.
+ */
+public class DownloadController {
+
+    final Logger API_REQUEST_LOG = LoggerFactory.getLogger(AppLoggerConstants.API_METRICS_LOGGER);
+    final AppLoggerUtil loggerUtil = AppLoggerUtil.getInstance();
+    final AppLoggerUtil appLoggerUtil = loggerUtil;
```
Comment (ash3rz) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/file/DownloadController.java]: I see this was refactored from `FileDownloadController`, but is there a reason why we had two `AppLoggerUtil`s?

Comment (sri2k1us) [inline ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/file/DownloadController.java]: I had the same question. I will check with @strootman and make changes if necessary!

File: ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java [hunk 3]
```diff
@@ -696,4 +707,40 @@ public void onSelect(SelectEvent event) {
...
+    @Override
+    public void onDownloadClick(TemplateDownloadEvent event) {
+        final String encodedSimpleDownloadURL =
+                diskResourceService.downloadTemplate(event.getSelectedTemplateId());
+        WindowUtil.open(encodedSimpleDownloadURL, "width=100,height=100");
+    }
+
+    @Override
+    public void onDownloadTemplateSelected(DownloadTemplateSelectedEvent event) {
+        final MetadataView.Presenter.Appearance appearance =
```
Comment (ash3rz) [inline ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java]: If possible, we should add a test for this new method.

**Comment by ash3rz** [comment]
I had some minor comments, but looks good to me :shipit:

**Pull request merged**